connection-cache/src/connection_cache.rs

```diff
@@ -47,7 +48,7 @@ pub struct ConnectionCache<
 > {
     name: &'static str,
     map: Arc<RwLock<IndexMap<SocketAddr, /*ConnectionPool:*/ R>>>,
-    connection_manager: Arc<S>,
+    connection_manager: Arc<RwLock<S>>,
```

Comment:
Agreed that this is not ideal. Since the map is always locked when connection_manager is locked (but not vice versa), it should be possible to just use 1 lock to protect both. However, while I have tried to remove this lock, I have not been able to make Rust happy (non-buiilding code at https://github.com/ryleung-solana/solana/tree/quic-update-identity-test if you have suggestions).